eZ/Publish/API/Repository/URLWildcardService.php

```diff
@@ -75,4 +75,15 @@ public function loadAll($offset = 0, $limit = -1);
      * @return \eZ\Publish\API\Repository\Values\Content\URLWildcardTranslationResult
      */
     public function translate($url);
+
+    /**
+     * Performs lookup for the given url.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     *
+     * @param string $url
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\URLWildcard
+     */
+    public function lookup($url);
```

Comment:
I'm unsure why we need a new API.
The load all issue with `translate` is a implementation issue, and not a design issue right?

As in, we can:
- Add new method in SPI to do the lookup more effectively and cache that
- Use that in translate()

That said, as discussed on slack, even legacy loads all kind of, as in it generates a set of php files based that it caches to do the matching in pure php, but quite effectively:
https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/classes/ezurlwildcard.php#L217

Reply:
That's right. I've removed that new API method, kept it only for SPI handler. The method is used only in `UrlWildcardService::translate()`.

> Add new method in SPI to do the lookup more effectively and cache that

The SPIs `lookup` method is cached, please refer to: https://github.com/ezsystems/ezpublish-kernel/pull/2609/files#diff-485676a9318fdeefc018635a35f8a7faR77

I'll look for some good approach for `loadAll()`.